Sources/PointFree/Account/UpdateProfile.swift

```diff
@@ -4,26 +4,74 @@ import HttpPipeline
 import Prelude
 import Tuple
 
-public struct ProfileData: Codable {
+// NB: remove this `Encodable` to get a runtime crash
+public struct ProfileData: Encodable {
   public let email: EmailAddress
   public let name: String
+  public let emailSettings: [String: String]
+
+  public enum CodingKeys: String, CodingKey {
+    case email
+    case name
+    case emailSettings
+  }
+}
+
+extension ProfileData: Decodable {
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+
+    self.email = try container.decode(EmailAddress.self, forKey: .email)
+    self.name = try container.decode(String.self, forKey: .name)
+    self.emailSettings = (try? container.decode([String: String].self, forKey: .emailSettings)) ?? [:]
```

Comment:
had to provide custom decode `init` so that if `emailSettings` is missing I can default to `[:]`. I think it'd be better to have `emailSettings: [String: String]?` but the route wasnt working for me